libs/libskynet/src/types.ts

```diff
@@ -0,0 +1,68 @@
+// dataFn can take any object as input and has no return value. The input is
+// allowed to be undefined.
+type dataFn = (data?: any) => void;
+
+// error is an error type that is either a string or a null. If the value is
+// null, that indicates that there was no error. If the value is a string, it
+// indicates that there was an error and the string contains the error message.
+//
+// The skynet libraries prefer this error type to the standard Error type
+// because many times skynet libraries need to pass errors over postMessage,
+// and the 'Error' type is not able to be sent over postMessage.
```

Comment:
that's not true, classic `Error` serialises into a string just fine and it drops the stacktrace when it does:

```js
const error = new Error("some string explanation")
...
error.toString(); // 'Error: some string explanation'
```

Reply:
then you have to explicitly call `toString` on it, you can't just pass an `Error` over postmessage, the copy function fails.

Reply:
> then you have to explicitly call `toString` on it, you can't just pass an `Error` over postmessage, the copy function fails.

`postMessage` works just fine with Error object though:

```js
window.addEventListener("message", (event) => {
    console.log(event.data)
    console.log(event.data.foo.message); // prints "bar baz"
}, false);
postMessage({ foo: new Error("bar baz") });
```

Reply:
also, `Error` types don't have good `toString` methods you have to use use `.message` if you want it to be comprehensible.